cpp/unreal_plugins/SpEngine/Source/SpEngine/GameWorldService.h

```diff
@@ -178,8 +368,651 @@ class GameWorldService {
+                spawn_params.bAllowDuringConstructionScript = sp_spawn_params.getObj().bAllowDuringConstructionScript;
+                spawn_params.NameMode = static_cast<FActorSpawnParameters::ESpawnActorNameMode>(sp_spawn_params.getObj().NameMode);
+                spawn_params.ObjectFlags = static_cast<EObjectFlags>(sp_spawn_params.getObj().ObjectFlags);
+
+                return reinterpret_cast<uint64_t>(
+                    UnrealClassRegistrar::spawnActor(
+                        class_name,
+                        world_,
+                        FVector(location.at(0), location.at(1), location.at(2)),
+                        FRotator(rotation.at(0), rotation.at(1), rotation.at(2)),
+                        spawn_params));
+            });
+
+        unreal_entry_point_binder->bindFuncUnreal("game_world_service", "create_component_outside_owner_constructor", 
+            [this](const std::string& class_name, const uint64_t& owner, const std::string& name) -> uint64_t {
+                return reinterpret_cast<uint64_t>(UnrealClassRegistrar::createComponentOutsideOwnerConstructor(class_name, reinterpret_cast<AActor*>(owner), name));
+            });
+
+        unreal_entry_point_binder->bindFuncUnreal("game_world_service", "create_scene_component_outside_owner_constructor_from_actor",
+            [this](const std::string& class_name, const uint64_t& actor, const std::string& name) -> uint64_t {
+                return reinterpret_cast<uint64_t>(UnrealClassRegistrar::createSceneComponentOutsideOwnerConstructor(class_name, reinterpret_cast<AActor*>(actor), name));
+            });
+
+        unreal_entry_point_binder->bindFuncUnreal("game_world_service", "create_scene_component_outside_owner_constructor_from_object",
+            [this](const std::string& class_name, const uint64_t& owner, const uint64_t& parent, const std::string& name) -> uint64_t {
+                return reinterpret_cast<uint64_t>(
+                    UnrealClassRegistrar::createSceneComponentOutsideOwnerConstructor(class_name, reinterpret_cast<UObject*>(owner), reinterpret_cast<USceneComponent*>(parent), name));
+            });
+
+        unreal_entry_point_binder->bindFuncUnreal("game_world_service", "create_scene_component_outside_owner_constructor_from_component",
+            [this](const std::string& class_name, const uint64_t& owner, const std::string& name) -> uint64_t {
+                return reinterpret_cast<uint64_t>(UnrealClassRegistrar::createSceneComponentOutsideOwnerConstructor(class_name, reinterpret_cast<USceneComponent*>(owner), name));
+            });
+
+        unreal_entry_point_binder->bindFuncUnreal("game_world_service", "new_object",
+            [this](
+                const std::string& class_name,
+                const uint64_t& outer,
+                const std::string& name,
+                const std::map<std::string, std::string>& unreal_obj_strings,
```

Comment:
Let's create the convention that `unreal_obj_strings` is always the last parameter (unless there are enum flags, in which case we can have an additional parameter after that of type `const std::map<std::string, std::vector<std::string>>&` called `unreal_flag_strings`.

Reply:
Agreed. In our current functions where we use `unreal_flag_strings`, we do not need an `unreal_obj_strings` argument.